test-harness/src/test/groovy/com/netflix/counductor/integration/test/ForkJoinSpec.groovy

```diff
@@ -0,0 +1,874 @@
+package com.netflix.counductor.integration.test
+
+import com.netflix.archaius.guice.ArchaiusModule
+import com.netflix.conductor.common.metadata.tasks.Task
+import com.netflix.conductor.common.metadata.tasks.TaskDef
+import com.netflix.conductor.common.run.Workflow
+import com.netflix.conductor.core.execution.WorkflowExecutor
+import com.netflix.conductor.core.execution.tasks.SubWorkflow
+import com.netflix.conductor.service.ExecutionService
+import com.netflix.conductor.service.MetadataService
+import com.netflix.conductor.test.util.WorkflowTestUtil
+import com.netflix.conductor.tests.utils.TestModule
+import com.netflix.governator.guice.test.ModulesForTesting
+import spock.lang.Shared
+import spock.lang.Specification
+
+import javax.inject.Inject
+
+@ModulesForTesting([TestModule.class, ArchaiusModule.class])
+class ForkJoinSpec extends Specification {
+
+    @Inject
+    ExecutionService workflowExecutionService
+
+    @Inject
+    MetadataService metadataService
+
+    @Inject
+    WorkflowExecutor workflowExecutor
+
+    @Inject
+    WorkflowTestUtil workflowTestUtil
+
+    @Shared
+    def FORK_JOIN_WF = 'FanInOutTest'
+
+    @Shared
+    def FORK_JOIN_NESTED_WF = 'FanInOutNestedTest'
+
+    @Shared
+    def FORK_JOIN_NESTED_SUB_WF = 'FanInOutNestedSubWorkflowTest'
+
+    @Shared
+    def WORKFLOW_FORK_JOIN_OPTIONAL_SW = "integration_test_fork_join_optional_sw"
+
+    def cleanup() {
+        workflowTestUtil.clearWorkflows()
+    }
+
+    def setup() {
+        workflowTestUtil.registerWorkflows('fork_join_integration_test.json',
+                'fork_join_with_no_task_retry_integration_test.json',
+                'nested_fork_join_integration_test.json',
+                'simple_workflow_1_integration_test.json',
+                'nested_fork_join_with_sub_workflow_integration_test.json',
+                'simple_one_task_sub_workflow_integration_test.json',
+                'fork_join_with_optional_sub_workflow_forks_integration_test.json'
+        )
+    }
+
+    /**
+     *               start
+     *                 |
+     *               fork
+     *              /    \
+     *         task1     task2
+     *          \        /
+     *          task3   /
+     *           \     /
+     *            \  /
+     *            join
+     *              |
+     *             task4
+     *              |
+     *             End
+     */
+    def "Test a simple workflow with fork join success flow"() {
+        setup: "Ensure that all the tasks involved in the workflow have a retry count of 0"
```

Comment:
I believe this was done to maintain consistency with the existing JUnit tests, but this spec does not use/need modification of retry counts, so this could be removed.

Reply:
This was done before using the `fork_join_workflow_without_retryable_tasks` , this test can be changed to use a different workflow and the `setup` and `cleanup` are not needed.